### PR TITLE
500px client

### PR DESCRIPTION
**Pull request by ak-14**

### CODE REVIEW

File: lib/pixelfinder/image_client/500px_client/500px_client.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      def build_image(photo)
+        return {} unless exif_has_required_tags?(photo)
+
+        image = format_exif(photo)
+
+        creator = photo['user']['fullname'].presence || photo['user']['username']
+        image[:creator] = creator
+
+        image[:page_url] = "https://500px/photo/#{photo['id']}"
+        #image[:page_url] = photo['url']
```
Comment (iFahad7) [inline lib/pixelfinder/image_client/500px_client/500px_client.rb]: remove comments

File: lib/pixelfinder/image_client/500px_client/500px_client.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        urls = photo['images']
+        urls.each do |url|
+          image[:image_url] = url['url'] if url['size'] == 1080
```
Comment (iFahad7) [inline lib/pixelfinder/image_client/500px_client/500px_client.rb]: What are theses number 1080, 30. If they have special meaning, they should be in a constant

File: lib/pixelfinder/image_client/500px_client/500px_client.rb [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        tags = ['camera', 'lens', 'shutter_speed', 'aperture',
+                'iso', 'focal_length', 'user']
+        tags.any? { |k| exif.key?(k) && exif[k].present? }
+      end
+
+    end
+  end
+end
+
+#def get_access_token
```
Comment (iFahad7) [inline lib/pixelfinder/image_client/500px_client/500px_client.rb]: Remove comments

File: lib/pixelfinder/image_client/500px_client/500px_client.rb [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+BASE_URL = 'https://api.500px.com'
+
+module PixelFinder
+  module ImageClient
+
+    class FiveHundredPxClient < ImageSourceClient
+
+      def initialize
+        p "get_access_token: Initializing Consumer"
```
Comment (iFahad7) [inline lib/pixelfinder/image_client/500px_client/500px_client.rb]: Use a logger like a proper software engineer bro 🤓 

Check out section 2 in http://guides.rubyonrails.org/debugging_rails_applications.html

File: lib/pixelfinder/image_client/500px_client/500px_client.rb [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+require 'rubygems'
+require 'oauth'
+require 'multi_json'
+
+CONSUMER_KEY = Rails.application.secrets.five00px_consumer_key
+CONSUMER_SECRET = Rails.application.secrets.five00px_consumer_secret
+USERNAME = Rails.application.secrets.five00px_username
+PASSWORD = Rails.application.secrets.five00px_password
+
+BASE_URL = 'https://api.500px.com'
```
Comment (iFahad7) [inline lib/pixelfinder/image_client/500px_client/500px_client.rb]: Usaully it is better to put constants in the namespace they belong to, in this case, it is `FiveHundredPxClient` class

File: lib/pixelfinder/image_client/500px_client/500px_client.rb [hunk 6]
```diff
@@ -0,0 +1,108 @@
+require 'pixelfinder/image_client/image_source_client'
+require 'rubygems'
+require 'oauth'
+require 'multi_json'
+
+CONSUMER_KEY = Rails.application.secrets.five00px_consumer_key
```
Comment (iFahad7) [inline lib/pixelfinder/image_client/500px_client/500px_client.rb]: We already have they values accessible in `Rails.application.secrets`, why put them in a constant?

File: config/secrets.yml [hunk 7]
```diff
@@ -14,15 +14,27 @@ development:
   secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
   flickr_api_key: <%= ENV["FLICKR_API_KEY"] %>
   flickr_shared_secret: <%= ENV["FLICKR_SHARED_SECRET"] %>
+  five00px_consumer_key: <%= ENV["500PX_CONSUMER_KEY"] %>
```
Comment (iFahad7) [inline config/secrets.yml]: why not _500px? I think it is better than five00px

**Comment by ak-14** [comment]
Thanks for the feedback bro it's very helpful.

**Pull request merged**